internal/services/iotoperations/iotoperations_broker_listener_resource_test.go

```diff
@@ -0,0 +1,255 @@
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		{
+			Config:      r.requiresImport(data),
+			ExpectError: acceptance.RequiresImportError("azurerm_iotoperations_broker_listener"),
+		},
+	})
+}
+
+func TestAccIotOperationsBrokerListener_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_iotoperations_broker_listener", "test")
+	r := IotOperationsBrokerListenerResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccIotOperationsBrokerListener_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_iotoperations_broker_listener", "test")
+	r := IotOperationsBrokerListenerResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func (IotOperationsBrokerListenerResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	// Parse the ID to get the structured ID object, not individual strings
+	id, err := brokerlistener.ParseListenerID(state.ID)
+	if err != nil {
+		return nil, fmt.Errorf("parsing %s: %+v", state.ID, err)
+	}
+
+	// Use the parsed ID object in the Get call
+	resp, err := clients.IoTOperations.BrokerListenerClient.Get(ctx, *id)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving %s: %+v", state.ID, err)
+	}
+
+	return utils.Bool(resp.Model != nil), nil
+}
+
+// template builds the minimal provider + resource_group; NOTE: you must create an IoT Operations instance and broker
+func (IotOperationsBrokerListenerResource) template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-iotops-%d"
+  location = "%s"
+}
+`, data.RandomInteger, data.Locations.Primary)
+}
+
+func (r IotOperationsBrokerListenerResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+# TODO: create or reference an IoT Operations instance and broker here.
+
+resource "azurerm_iotoperations_broker_listener" "test" {
+  name                = "acctest-bl-%s"
+  resource_group_name = azurerm_resource_group.test.name
+
+  # Replace with the actual instance and broker names created in this template:
+  instance_name = "REPLACE_WITH_INSTANCE_NAME"
+  broker_name   = "REPLACE_WITH_BROKER_NAME"
```

Comment:
is there a reason we use `REPLACE_WITH_XXX` here? can we use a environment variable, or add a comment here about hot to setup them? the same question applies to the `complete` config